app/src/test/java/com/github/se/travelpouch/ui/notification/NotificationScreenUnitTest.kt

```diff
@@ -0,0 +1,168 @@
+                senderUid = senderUid,
+                receiverUid = receiverUid,
+                travelUid = travel1Uid,
+                content = content1,
+                notificationType = notificationType1,
+                timestamp = now))
+    val result = categorizeNotifications(notifications)
+    assertEquals(1, result["This week"]?.size)
+  }
+
+  @Test
+  fun categorizeNotifications_lastWeek() {
+    val now = Timestamp(Calendar.getInstance().apply { add(Calendar.WEEK_OF_YEAR, -1) }.time)
+    val notifications =
+        listOf(
+            Notification(
+                notificationUid = notificationUid,
+                senderUid = senderUid,
+                receiverUid = receiverUid,
+                travelUid = travel1Uid,
+                content = content1,
+                notificationType = notificationType1,
+                timestamp = now))
+    val result = categorizeNotifications(notifications)
+    assertEquals(1, result["Last week"]?.size)
+  }
+
+  @Test
+  fun categorizeNotifications_lastMonth() {
+    val now = Timestamp(Calendar.getInstance().apply { add(Calendar.MONTH, -1) }.time)
+    val notifications =
+        listOf(
+            Notification(
+                notificationUid = notificationUid,
+                senderUid = senderUid,
+                receiverUid = receiverUid,
+                travelUid = travel1Uid,
+                content = content1,
+                notificationType = notificationType1,
+                timestamp = now))
+    val result = categorizeNotifications(notifications)
+    assertEquals(1, result["Last month"]?.size)
+  }
+
+  @Test
+  fun categorizeNotifications_lastYear() {
+    val now = Timestamp(Calendar.getInstance().apply { add(Calendar.YEAR, -1) }.time)
+    val notifications =
+        listOf(
+            Notification(
+                notificationUid = notificationUid,
+                senderUid = senderUid,
+                receiverUid = receiverUid,
+                travelUid = travel1Uid,
+                content = content1,
+                notificationType = notificationType1,
+                timestamp = now))
+    val result = categorizeNotifications(notifications)
+    assertEquals(1, result["Last year"]?.size)
+  }
+
+  @Test
+  fun categorizeNotifications_emptyList() {
+    val notifications = emptyList<Notification>()
+    val result = categorizeNotifications(notifications)
+    assertEquals(0, result["This week"]?.size)
+    assertEquals(0, result["Last week"]?.size)
+    assertEquals(0, result["Last month"]?.size)
+    assertEquals(0, result["Last year"]?.size)
+  }
+
+  @Test
+  fun isThisWeek_test() {
+    val now = Calendar.getInstance()
+    var date = now.time
+    assertTrue(isThisWeek(date, now))
+
+    now.add(Calendar.WEEK_OF_YEAR, -1)
+    date = now.time
+    assertFalse(isThisWeek(date, Calendar.getInstance()))
+
+    now.add(Calendar.YEAR, -1)
+    date = now.time
+    assertFalse(isThisWeek(date, Calendar.getInstance()))
+  }
```

Comment:
To have a better test I would not substract 1 year in your second add, but rather one week again as it is closer to the current week